speeduino/scheduler.h

```diff
@@ -117,167 +135,51 @@ struct Schedule {
   void (*EndFunction)();          ///< End Callback function for schedule
 
   //pure virtual functions, these are really defined in sub-classes
-  virtual COMPARE_TYPE getCounter(void)=0; //Function for getting counter value
-  virtual void setCompare(COMPARE_TYPE compareValue)=0; //Function for setting counter compare value
-  virtual void timerDisable(void)=0; //Function to disable timer for specific channel
-  virtual void timerEnable(void)=0; //Function to enable timer for specific channel
+  COMPARE_TYPE getCounter(void) { return counter; }; //Function for getting counter value
+  void setCompare(COMPARE_TYPE compareValue) { compare=(COMPARE_TYPE)compareValue; }; //Function for setting counter compare value
```

Comment:
```suggestion
  void setCompare(COMPARE_TYPE compareValue) { compare=compareValue; }; //Function for setting counter compare value
```
Or should we cast `compare=(compare_t)compareValue` here instead? Plain version without casts at all looks of cource nicest of them all...

Reply:
> Or should we cast compare=(compare_t)compareValue here instead?

I don't think that cast will work - compare_t is a reference (volatile uint16_t &) on Mega 2560,

Reply:
Yeah, in this case it is probably good to leave it without the cast because compareValue is already COMPARE_TYPE. Then also the original operation is not changed. I remember STM32 had some timers 32bit and some 16bit, have to check what was the thing there, and if all timers registers are accessible same way using the references as they may be also different type...